common/thorhelper/thorcommon.cpp

```diff
@@ -1823,14 +1823,26 @@ void setAutoAffinity(unsigned curProcess, unsigned processPerMachine, const char
     unsigned numaMap[NUMA_NUM_NODES];
     unsigned numNumaNodes = 0;
 #if defined(LIBNUMA_API_VERSION) && (LIBNUMA_API_VERSION>=2)
-    for (unsigned i=0; i<=numa_max_node(); i++)
+    unsigned maxcpus = numa_num_configured_cpus();
```

Comment:
I used this because I want to go through all cpus.  (Although it does assume that cpus are contiguously numbered).  If you used num_task_cpus it may return a low number of cpus - all of which may be disabled.  The whole api is a bit of a mess.